cmake/CopyRuntimeDependencies.cmake

```diff
@@ -0,0 +1,186 @@
+function(copy_runtime_dependencies TARGET)
+  set(valid_options "DEPENDENCY;LIBRARY")
+  foreach(arg IN LISTS ARGN)
+    if("${arg}" IN_LIST valid_options)
+      set(mode "${arg}")
+    elseif("${mode}" STREQUAL "DEPENDENCY")
+      list(LENGTH RUNTIME_DEPENDENCY_SOURCES current_dependencies)
+      list(APPEND RUNTIME_DEPENDENCY_SOURCES "${arg}")
+    elseif("${mode}" STREQUAL "LIBRARY")
+      list(APPEND current_libs "${arg}")
+    endif()
+  endforeach()
+  get_property(multi_config GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+  if(RUNTIME_DEPENDENCY_SOURCES)
+    foreach(dependency_idx RANGE 0 ${current_dependencies})
+      list(GET RUNTIME_DEPENDENCY_SOURCES ${dependency_idx} dependency)
+      if(WIN32)
+        add_custom_command(
+        TARGET ${TARGET} POST_BUILD
+        COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                $<TARGET_RUNTIME_DLLS:${dependency}> $<TARGET_FILE_DIR:${TARGET}>
+        COMMAND_EXPAND_LISTS
+        )
+      elseif(APPLE)
+        get_target_property(libs ${dependency} LINK_LIBRARIES)
+        if(libs)
+          foreach(lib_idx ${libs})
+            if("${lib_idx}" MATCHES "\<.*")
+              break()
+            else()
+              get_target_property(type ${lib_idx} TYPE)
+              if("${type}" STREQUAL "INTERFACE_LIBRARY" OR "$type" STREQUAL "STATIC_LIBRARY")
+                break()
+              endif()
+              if("${lib_idx}" MATCHES ".*::.*")
+                string(REGEX REPLACE ".*::" "" lib_safe_name "${lib_idx}")
+              else()
+                set(lib_safe_name "${lib_idx}")
+              endif()
+              list(APPEND dep_lib_snippet
+                "
+                if(EXISTS otool.txt)
```

Comment:
What is all this doing?

Reply:
_Heh._ :sans: 
Currently, this will:
- Run `otool` on the provided target library/app with `grep` to find the RPATH entry that was linked for the dependency
- Then it will use `install_name_tool` (IIRC, on mobile atm) and update said library/app with a new RPATH for the dependency

I did this so that on load, if _for some reason the samples get moved_, it knows where to look inside the App Bundles for the dependent libs. This is specifically Apple-only, Linux doesn't need this and Windows just uses CMake's support for identifying runtime DLLs.